python/tests/test_logical_record.py

```diff
@@ -113,7 +113,20 @@ def test_absent_attribute_in_template(tmpdir, merge_files_oneLR):
         obj = f.object('VERY_MUCH_TESTY_SET', 'OBJECT', 1, 1)
         assert obj.attic['DEFAULT_ATTRIBUTE']
 
+@pytest.mark.future_warning_repcode_different_no_value
+def test_repcode_different_no_value(tmpdir, merge_files_oneLR):
+    path = os.path.join(str(tmpdir), 'different-repcode-no-value.dlis')
+    content = [
+        'data/chap3/start.dlis.part',
+        'data/chap3/template/default.dlis.part',
+        'data/chap3/object/object.dlis.part',
+        'data/chap3/objattr/csingle-novalue.dlis.part'
+    ]
+    merge_files_oneLR(path, content)
 
+    with dlisio.load(path) as (f, *_):
+        obj = f.object('VERY_MUCH_TESTY_SET', 'OBJECT', 1, 1)
+        assert obj.attic['DEFAULT_ATTRIBUTE']
```

Comment:
OK, there is one problem with this.
We do not report an error anymore, which means that
`obj.attic['DEFAULT_ATTRIBUTE'] `equals something. And we want this something to make sense, so we have to check it.

Right now in the code it equals [-0.75, 10.0] because this is the value in the template (float). But we changed the type from float to csingle, we shouldn't be seeing this value anymore!
in `parse.cpp/object_vector parse_objects` we use `patch_missing_value( attr.value, count, attr.reprc )` after reporting an error. Problem is that `patch_missing_value` does nothing if there already was a value set in template. It then leaves it to be an old one. In case of the test, floats.

What we want in case of such error is to disregard whatever is in the template and create a default value of new type.
You can do it by adding
`attr.value = mpark::monostate{};`
right after reporting the warning. This line says "ok, I declare that I have no value whatsoever", thus allowing us to supply it with default one in `patch_missing_value`.
Then the check in the test would be correct to the test logic
`assert obj.attic['DEFAULT_ATTRIBUTE'] == [0j, 0j]`